sklearn/manifold/t_sne.py

```diff
@@ -413,7 +413,7 @@ def _fit(self, X):
             If the metric is 'precomputed' X must be a square distance
             matrix. Otherwise it contains a sample per row.
         """
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'], dtype=np.float)
```

Comment:
I would rather use `dtype=np.float64` explicitly here. `np.float` is an alias for the Python `float` builtin and the precision could in theory be platform specific although I think it maps to 64bit floats on all supported platforms. Anyway let's be explicit rather than implicit.